### PR TITLE
Add additional entities unit tests

### PR DESCRIPTION
**Pull request by PedramNavid**
This adds more comprehensive tests on entities to help increase coverage, plus a test of the duplicate removal method in the spec loader.

### CODE REVIEW

File: src/permifrost/core/permissions/snowflake_spec_loader.py [hunk 1]
```diff
@@ -465,7 +465,8 @@ def process_users(self, generator, entity_type, entity_name, config):
         )
         return sql_commands
 
-    def remove_duplicate_queries(self, sql_commands: List[Dict]) -> List[Dict]:
+    @staticmethod
+    def remove_duplicate_queries(sql_commands: List[Dict]) -> List[Dict]:
```
Comment (blake-enyart) [inline src/permifrost/core/permissions/snowflake_spec_loader.py:469]: Could you clarify the move to a static method here for me?

File: tests/permifrost/core/permissions/test_entities.py [hunk 2]
```diff
@@ -26,12 +26,36 @@ def entities(test_dir):
...
 class TestEntityGenerator:
     def test_entity_databases(self, entities):
-        expected = {"demo"}
+        expected = {"demo", "shared_demo"}
         assert entities["databases"] == expected
 
-    def test_entity_roles(self, entities):
-        expected = {"accountadmin", "demo", "securityadmin", "sysadmin", "useradmin"}
+    def test_entity_require_owner(self, entities):
+        assert entities["require_owner"] is True
```
Comment (blake-enyart) [inline tests/permifrost/core/permissions/test_entities.py:33]: Will this cause a breakage with the `require-owner` key?

File: src/permifrost/core/permissions/snowflake_spec_loader.py (hunk 1 above)
Comment (PedramNavid) [reply src/permifrost/core/permissions/snowflake_spec_loader.py:469]: Static method means we can use this method without having to instantiate the class, since the method doesn't actually rely on anything within the class itself. 

It's nice because then I can test the function without mocking a connection or a file for the class by doing this.
```
SnowflakeSpecLoader.remove_duplicate_queries(...)
```

instead of 

```
loader = SnowflakeSpecLoader(a_bunch_of_mocked_things)
loader.remove_duplicate_queries
```
